### ISSUE TITLE
Hyperparameter Tuning

### ISSUE THREAD

**Issue by rajithalder-pronto** [ultralytics/ultralytics#23677]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Hey so if i dont want some hyperparameter to not be tuned i just make the search space (0,0) right?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @rajithalder-pronto, thank you for your interest in Ultralytics 🚀! This is an automated response to help you get unstuck quickly—an Ultralytics engineer will also assist soon 🙌. We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question (including hyperparameter tuning), please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) 📌.

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

To help the team answer your hyperparameter-tuning question precisely 🧪, please share:
-Your exact command or Python snippet used for tuning (and how you defined the search space)
-Your `ultralytics` version (`pip show ultralytics`) and environment details (OS, Python, PyTorch, CUDA)
-The model and task you’re tuning, plus any relevant training logs

If you think something is not behaving as expected (possible 🐛), please include a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) that we can run end-to-end.

**Comment by Y-T-G** [comment]
You just don't include them in the search space dict.

**Comment by rajithalder-pronto** [comment]
> You just don't include them in the search space dict.

Oh Okay. Thanks

**Comment by glenn-jocher** [comment]
Yep—don’t use `(0, 0)` (that can break bounds/mutation); to keep a hyperparameter fixed, leave it out of `space` and (if needed) pass the fixed value directly to `model.tune(...)` (e.g. `batch=16`), as shown in the [Hyperparameter Tuning guide](https://docs.ultralytics.com/guides/hyperparameter-tuning/).

**Comment by rajithalder-pronto** [comment]
> Yep—don’t use `(0, 0)` (that can break bounds/mutation); to keep a hyperparameter fixed, leave it out of `space` and (if needed) pass the fixed value directly to `model.tune(...)` (e.g. `batch=16`), as shown in the [Hyperparameter Tuning guide](https://docs.ultralytics.com/guides/hyperparameter-tuning/).

Thats exactly what i was gonna ask. Thank you Glenn and Y-T-G as always

**Comment by glenn-jocher** [comment]
Glad it helped—credit to the whole YOLO/Ultralytics community; if anything else comes up while tuning (like which fixed `train()` args can be forwarded into `model.tune()` or how resume behaves), just follow up here or open a new issue, and the [Model.tune() reference](https://docs.ultralytics.com/reference/engine/model/#ultralytics.engine.model.Model.tune) is the canonical list of supported arguments.
